// ble-central-advertisements/gap/gap-attribute-code.js
/**
 * https://www.bluetooth.com/specifications/assigned-numbers/generic-access-profile/
 */
export var GapAttributeCode;
(function (GapAttributeCode) {
    GapAttributeCode[GapAttributeCode["FLAGS"] = 1] = "FLAGS";
    GapAttributeCode[GapAttributeCode["INCOMPLETE_LIST_OF_SERVICE_UUIDS_16_BIT"] = 2] = "INCOMPLETE_LIST_OF_SERVICE_UUIDS_16_BIT";
    GapAttributeCode[GapAttributeCode["COMPLETE_LIST_OF_SERVICE_UUIDS_16_BIT"] = 3] = "COMPLETE_LIST_OF_SERVICE_UUIDS_16_BIT";
    GapAttributeCode[GapAttributeCode["INCOMPLETE_LIST_OF_SERVICE_UUIDS_32_BIT"] = 4] = "INCOMPLETE_LIST_OF_SERVICE_UUIDS_32_BIT";
    GapAttributeCode[GapAttributeCode["COMPLETE_LIST_OF_SERVICE_UUIDS_32_BIT"] = 5] = "COMPLETE_LIST_OF_SERVICE_UUIDS_32_BIT";
    GapAttributeCode[GapAttributeCode["INCOMPLETE_LIST_OF_SERVICE_UUIDS_128_BIT"] = 6] = "INCOMPLETE_LIST_OF_SERVICE_UUIDS_128_BIT";
    GapAttributeCode[GapAttributeCode["COMPLETE_LIST_OF_SERVICE_UUIDS_128_BIT"] = 7] = "COMPLETE_LIST_OF_SERVICE_UUIDS_128_BIT";
    GapAttributeCode[GapAttributeCode["LOCAL_NAME_SHORTENED"] = 8] = "LOCAL_NAME_SHORTENED";
    GapAttributeCode[GapAttributeCode["LOCAL_NAME_COMPLETE"] = 9] = "LOCAL_NAME_COMPLETE";
    GapAttributeCode[GapAttributeCode["TX_POWER_LEVEL"] = 10] = "TX_POWER_LEVEL";
    GapAttributeCode[GapAttributeCode["DEVICE_CLASS"] = 13] = "DEVICE_CLASS";
    GapAttributeCode[GapAttributeCode["SIMPLE_PAIRING_HASH_C_192"] = 14] = "SIMPLE_PAIRING_HASH_C_192";
    GapAttributeCode[GapAttributeCode["SIMPLE_PAIRING_RANDOMIZER_R_192"] = 15] = "SIMPLE_PAIRING_RANDOMIZER_R_192";
    GapAttributeCode[GapAttributeCode["DEVICE_ID"] = 16] = "DEVICE_ID";
    GapAttributeCode[GapAttributeCode["SECURITY_MANAGER_OUT_OF_BAND_FLAGS"] = 17] = "SECURITY_MANAGER_OUT_OF_BAND_FLAGS";
    GapAttributeCode[GapAttributeCode["SLAVE_CONNECTION_INTERVAL_RANGE"] = 18] = "SLAVE_CONNECTION_INTERVAL_RANGE";
    GapAttributeCode[GapAttributeCode["LIST_OF_SERVICE_SOLICITATION_UUIDS_16_BIT"] = 20] = "LIST_OF_SERVICE_SOLICITATION_UUIDS_16_BIT";
    GapAttributeCode[GapAttributeCode["LIST_OF_SERVICE_SOLICITATION_UUIDS_128_BIT"] = 21] = "LIST_OF_SERVICE_SOLICITATION_UUIDS_128_BIT";
    GapAttributeCode[GapAttributeCode["SERVICE_DATA_UUID_16_BIT"] = 22] = "SERVICE_DATA_UUID_16_BIT";
    GapAttributeCode[GapAttributeCode["TARGET_ADDRESS_PUBLIC"] = 23] = "TARGET_ADDRESS_PUBLIC";
    GapAttributeCode[GapAttributeCode["TARGET_ADDRESS_RANDOM"] = 24] = "TARGET_ADDRESS_RANDOM";
    GapAttributeCode[GapAttributeCode["APPEARANCE"] = 25] = "APPEARANCE";
    GapAttributeCode[GapAttributeCode["ADVERTISING_INTERVAL"] = 26] = "ADVERTISING_INTERVAL";
    GapAttributeCode[GapAttributeCode["LE_BLUETOOTH_DEVICE_ADDRESS"] = 27] = "LE_BLUETOOTH_DEVICE_ADDRESS";
    GapAttributeCode[GapAttributeCode["LE_ROLE"] = 28] = "LE_ROLE";
    GapAttributeCode[GapAttributeCode["SIMPLE_PAIRING_HASH_C_256"] = 29] = "SIMPLE_PAIRING_HASH_C_256";
    GapAttributeCode[GapAttributeCode["SIMPLE_PAIRING_RANDOMIZER_R_256"] = 30] = "SIMPLE_PAIRING_RANDOMIZER_R_256";
    GapAttributeCode[GapAttributeCode["LIST_OF_SERVICE_SOLICITATION_UUIDS_32_BIT"] = 31] = "LIST_OF_SERVICE_SOLICITATION_UUIDS_32_BIT";
    GapAttributeCode[GapAttributeCode["SERVICE_DATA_UUID_32_BIT"] = 32] = "SERVICE_DATA_UUID_32_BIT";
    GapAttributeCode[GapAttributeCode["SERVICE_DATA_UUID_128_BIT"] = 33] = "SERVICE_DATA_UUID_128_BIT";
    GapAttributeCode[GapAttributeCode["LE_SECURE_CONNECTIONS_CONFIRMATION_VALUE"] = 34] = "LE_SECURE_CONNECTIONS_CONFIRMATION_VALUE";
    GapAttributeCode[GapAttributeCode["LE_SECURE_CONNECTIONS_RANDOM_VALUE"] = 35] = "LE_SECURE_CONNECTIONS_RANDOM_VALUE";
    GapAttributeCode[GapAttributeCode["URI"] = 36] = "URI";
    GapAttributeCode[GapAttributeCode["INDOOR_POSITIONING"] = 37] = "INDOOR_POSITIONING";
    GapAttributeCode[GapAttributeCode["TRANSPORT_DISCOVERY_DATA"] = 38] = "TRANSPORT_DISCOVERY_DATA";
    GapAttributeCode[GapAttributeCode["LE_SUPPORTED_FEATURES"] = 39] = "LE_SUPPORTED_FEATURES";
    GapAttributeCode[GapAttributeCode["CHANNEL_MAP_UPDATE_INDICATION"] = 40] = "CHANNEL_MAP_UPDATE_INDICATION";
    GapAttributeCode[GapAttributeCode["PB_ADV"] = 41] = "PB_ADV";
    GapAttributeCode[GapAttributeCode["MESH_MESSAGE"] = 42] = "MESH_MESSAGE";
    GapAttributeCode[GapAttributeCode["MESH_BEACON"] = 43] = "MESH_BEACON";
    GapAttributeCode[GapAttributeCode["BIG_INFO"] = 44] = "BIG_INFO";
    GapAttributeCode[GapAttributeCode["BROADCAST_CODE"] = 45] = "BROADCAST_CODE";
    GapAttributeCode[GapAttributeCode["INFORMATION_DATA_3D"] = 61] = "INFORMATION_DATA_3D";
    GapAttributeCode[GapAttributeCode["MANUFACTURER_SPECIFIC_DATA"] = 255] = "MANUFACTURER_SPECIFIC_DATA";
})(GapAttributeCode || (GapAttributeCode = {}));
